lib/summonerComparer.js

```diff
@@ -6,58 +6,182 @@ var championMap = require('../public/static/champions.json');
 
 //Takes in a list of tuples of Objects that contain the id of a champion and the summoner's stats for the particular champion
 exports.compareCommonChampions = function (commonChampions) {
-    var compareValue = 0;
+    var finalWinner = 0;
     var championCompareList = commonChampions.map(([championA, championB]) => {
-        var championCompare = compareChampionStats(championA.stats, championB.stats);
-        compareValue += championCompare.compare;
-        return ({
+
+        var winPercentage = compareWinPercentages(championA.stats, championB.stats);
+        var championKills = compareChampionKills(championA.stats, championB.stats);
+        var damageDealt = compareDamageDealt(championA.stats, championB.stats);
+        var creepScore = compareMinionsKilled(championA.stats, championB.stats);
+        var deaths = compareAverageDeaths(championA.stats, championB.stats);
+        var goldEarned = compareGoldEarned(championA.stats, championB.stats);
+        var turretsKilled = compareTurretsKilled(championA.stats, championB.stats);
+        var physicalDamage = comparePhysicalDamage(championA.stats, championB.stats);
+        var magicalDamage = compareMagicalDamage(championA.stats, championB.stats);
+        var assists = compareAssists(championA.stats, championB.stats);
+
+
+        var resultingWinner = winPercentage.winner + championKills.winner +
+            damageDealt.winner + creepScore.winner +
+            deaths.winner + goldEarned.winner +
+            turretsKilled.winner + physicalDamage.winner +
+            magicalDamage.winner + assists.winner;
+
+        finalWinner += (resultingWinner > 0 ? 1 : (resultingWinner < 0 ? -1 : 0));
+        return {
             championName: championMap[championA.id],
-            result: championCompare
-        });
+            winner: resultingWinner,
+            result: [winPercentage, championKills, damageDealt, creepScore,
+                deaths, goldEarned, turretsKilled, physicalDamage, magicalDamage, assists]
+        };
     });
 
-    var ret = {
-        compare: compareValue,  // overall result for all the champions.
+    return  {
+        finalWinner: finalWinner,  // overall result for all the champions.
         championCompareList: championCompareList // array will consist of json lke {id: championId, result: return value from compareChampionStats}
-    }
-    return ret;
+    };
+};
+
+
+var compareWinPercentages = function(summonerAStats, summonerBStats) {
```

Comment:
I know you merged this already but maybe the next time this code gets touched this could be cleaned up to just be one method/object that outputs the a result object given the stat they're comparing. All these methods basically do the same thing with slight parameter modification.